packages/fcl-core/src/transaction/transaction-error.ts

```diff
@@ -0,0 +1,21 @@
+import {FvmErrorCode, TransactionStatus} from "@onflow/typedefs"
+
+const ERROR_CODE_REGEX = /\[Error Code: (\d+)\]/
+
+export class TransactionError extends Error {
+  public code?: FvmErrorCode
+
+  private constructor(message: string, code?: FvmErrorCode) {
+    super(message)
+    this.code = code
+  }
+
+  static from(status: TransactionStatus): TransactionError | null {
+    if (status.statusCode === 0) return null
+
+    const match = status.errorMessage.match(ERROR_CODE_REGEX)
+    const code = match ? parseInt(match[1]) : undefined
```

Comment:
add radix to avoid converting string to OCT or HEX
```suggestion
    const code = match ? parseInt(match[1], 10) : undefined
```